src/mysql.erl

```diff
@@ -48,20 +50,53 @@
 -type server_reason() :: {Code :: integer(), SQLState :: binary() | undefined,
                           Message :: binary()}.
 
--type column_names() :: [binary()].
+-type column_name() :: binary().
+-type column_names() :: [column_name()].
+-type query() :: iodata().
 -type row() :: [term()].
 -type rows() :: [row()].
 
 -type query_filtermap_fun() :: fun((row()) -> query_filtermap_res())
                              | fun((column_names(), row()) -> query_filtermap_res()).
 -type query_filtermap_res() :: boolean()
                              | {true, term()}.
+-type query_filtermap() :: no_filtermap_fun | query_filtermap_fun().
```

Comment:
Sounds good.

I understand that as meaning we move everything from `query/5` into the new helper function, which accepts the `no_*` atoms in the respective places, and call that from `query/2,3,4,5`, where we drop the `no_*` values from the specs, but still allow them in the implmenetation in case someone uses them there (so we don't break anything that uses them, just generate dialyzer warnings). And similar for `execute/3,4,5`. Yes?